src/Service.Tests/CosmosTests/MutationTests.cs

```diff
@@ -252,62 +253,76 @@ public async Task MutationMissingRequiredPartitionKeyValueReturnError()
+        [DataRow("wildcard-exclude-fields-role", DataApiBuilderException.GRAPHQL_MUTATION_FIELD_AUTHZ_FAILURE)] // exclude wildcard
+        [DataRow("only-update-role", "The current user is not authorized to access this resource" )] // if create permission is not there
+        public async Task CreateItemWithAuthPermissions(string roleName, string expectedErrorMessage)
         {
             // Run mutation Add Earth;
             string id = Guid.NewGuid().ToString();
+            const string name = "test_name";
             string mutation = $@"
 mutation {{
-    createEarth (item: {{ id: ""{id}"" }}) {{
+    createEarth (item: {{ id: ""{id}"", name: ""{name}"" }}) {{
         id
+        name
     }}
 }}";
-            JsonElement response = await ExecuteGraphQLRequestAsync("createEarth", mutation, variables: new());
+            string authtoken = AuthTestHelper.CreateStaticWebAppsEasyAuthToken(specificRole: roleName);
+            JsonElement response = await ExecuteGraphQLRequestAsync("createEarth", mutation, variables: new(), authToken: authtoken, clientRoleHeader: roleName);
 
-            // Validate results
-            Assert.AreEqual(id, response.GetProperty("id").GetString());
+            // Validate the result contains the GraphQL authorization error code.
+            Console.WriteLine(response.ToString());
+            if (string.IsNullOrEmpty(expectedErrorMessage))
+            {
+                Assert.AreEqual(id, response.GetProperty("id").GetString());
+            }
+            else
+            {
+                // Validate the result contains the GraphQL authorization error code.
+                string errorMessage = response.ToString();
+                Assert.IsTrue(errorMessage.Contains(expectedErrorMessage));
+            }
         }
 
         /// <summary>
-        /// Mutation performed on the unauthorized fields throws permission denied error because the
-        /// field `name` is an excluded field for the create operation on the anonymous role defined
-        /// for entity 'earth'
+        /// Update Mutation performed on the fields with different auth permissions
+        /// It throws permission denied error if role doesn't have permission to perform the operation
         /// </summary>
         [TestMethod]
-        public async Task CreateItemWithUnauthorizedFieldsReturnsError()
+        [DataRow("field-mutation-with-read-permission", DataApiBuilderException.GRAPHQL_MUTATION_FIELD_AUTHZ_FAILURE)] // exclude and include fields
+        [DataRow("authenticated", null)] // full permission
+        [DataRow("only-update-role", "The mutation operation updateEarth was successful but the current user is unauthorized to view the response due to lack of read permissions")] // if only update permission is there
+        [DataRow("wildcard-exclude-fields-role", DataApiBuilderException.GRAPHQL_MUTATION_FIELD_AUTHZ_FAILURE)] // exclude wildcard
```

Comment:
can you clarify whether the test case with role == `wildcard-exclude-fields-role` means that the test will fail if the mutation input such as `id` and `name` below cause the authZ failure?:
```graphql
createEarth (item: {{ id: ""{id}"", name: ""{name}"" }}) {{
``` 

(Can also add test DisplayName to make test case self-explanatory)

Reply:
yes..you are right...because in `wildcard-exclude-fields-role` role, permission is limited to fields.